virtcontainers/sandbox.go

```diff
@@ -908,12 +908,18 @@ func (s *Sandbox) ListRoutes() ([]*vcTypes.Route, error) {
 }
 
 // startVM starts the VM.
-func (s *Sandbox) startVM() error {
+func (s *Sandbox) startVM() (err error) {
 	span, ctx := s.trace("startVM")
 	defer span.Finish()
 
 	s.Logger().Info("Starting VM")
 
+	defer func() {
+		if err != nil && s.hypervisor.pid() > 0 {
```

Comment:
I think it's better to register this cleanup function after startVM successfully, thus you shouldn't have to check the hypervisor's pid.

Reply:
I am not sure if it will happen that `s.hypervisor.startSandbox` return error but vm start, since it complicated when related to qemu, so I choose to locate cleanup before start